### ISSUE TITLE
Make drawings adapt to different handwritings

### ISSUE THREAD

**Issue by Wilker-a** [blastrock/kakugo#35]
hi :3 

i think this is possibly an expansion to #17, but i guess this is something for another issue.

an issue that i find myself into whenever i make the drawings is that at many times i have different ways of drawing the same letters, be they in the Latin Alphabet or the japanese Hiragana (which i am learning now :p).

that said, it would be nice to make the drawing lessons accept the strokes with any order and lenght, as long as it is considered legible enough. and with that, it would be also good to have different examples of legible letters to be shown when applicable (e.g the Hiragana letter "chi" is drawn as a lowercase "b" with a dash (ち), so why can't i draw "sa" as a "d" with a dashes (さ). also, how else it can be made, and can there be new ways of drawing that while keeping it clear and distinguishable?), as that can also help people read and iterate through different handwritings and typographies, just like we can read and associate the letter "J" written in cursive or in the simpler shape.

what do you think? :3

**Comment by blastrock** [comment]
When drawing Japanese characters, there is one correct order that you should learn, that's why kakugo ensures that you learn the correct one.

As for the different handwritings, I take my database from http://kanjivg.tagaini.net/ , and having just one handwriting is surely a lot of work for them, I don't think they will add more.

**Comment by Wilker-a** [comment]
why only a single order though? is it because of the meaning of the words, or just for the consistency of the handwriting (as most characters are mostly monospaced)?

and when the order is right, are the stroke counts and lenghts just a limitation of the library? e.g the characters わ, ね and れ looks like they could be done in 3 strokes, but you have to do in 2, while き and ぎ can be done in 3 and 5 but you have to make 1 more instead of being able to merge the diagonal dash and the curve into one

**Comment by OkyDooky** [comment]
The single order has to do with the culture. Their writing system originated from China, which also is very picky about how you do things. The process is just as important as the end result.
Japan just kept that tradition because they agreed with it and it stands pretty firm hundreds of years later (thousand plus?).

@blastrock So, if I wanted to address the issue that, say り is drawn in two strokes instead of the modern standard of one long one, I would have to bring that up with the tagaini.net devs and not here, correct?

**Comment by blastrock** [comment]
Yes, but even with that, it would imply work on kakugo to support having different writings for a single character.

**Comment by OkyDooky** [comment]
Understood. Though, my thinking was along the lines of having the VG database receiving the change and just using that updated one...but, again, not sure if/how that would affect implimentation on your end, so I won't assume.
Again, my thing was more a correction rather than a variation with no intention of supporting multiple versions. Which, I guess this isn't the right Issue to address that in, hence why I asked if I should even bring that up here or with the other devs.

**Comment by blastrock** [comment]
> Again, my thing was more a correction rather than a variation with no intention of supporting multiple versions.

Oh, I misunderstood then. If KanjiVG changes the drawings, it would be easy to update Kakugo's database with KanjiVG's last version indeed.

> Edit: I just looked at KanjiVG's Github and it looks like it's accumulated a lot of dust. So, not sure my small suggestion would get anywhere. :/

Yeah, the repository doesn't seem very active :(

> Thanks for your response. And keep up the great work! Love this app.

Thanks for your support! :)

**Comment by Wilker-a** [comment]
@OkyDooky thank you for explaining :3

@blastrock thank you for making this app, and thank you for making it [free](https://www.gnu.org/philosophy/free-sw.html). <3

**Comment by Wilker-a** [comment]
(i misclicked, sorry!)
